### ISSUE TITLE
custom docker socket for image scan option

### ISSUE THREAD

**Issue by aswath-s-tw** [aquasecurity/trivy#2997]
It would be great if we can have a flag for scanning local images with custom docker socket path.

I use colima as the container runtime for docker. The unix domain socket path for colima is `unix:///Users/<username>/.colima/default/docker.sock`

while doing `trivy image <image_name>`, it throws an error saying `Cannot connect to the Docker daemon at unix:///var/run/docker.sock`. Would be cool if it had an option similar to `trivy image --socket <socket_path>`.

**Comment by aswath-s-tw** [comment]
hey trivy team, please do review this request. Would like to work on it !

**Comment by titanventura** [comment]
Hey trivy team. Is this feature request even valid ? Please validate it so that I can start working on a solution.

**Comment by aswath-s-tw** [comment]
Ahh ! Never mind about this issue ! It is easily possible in colima too if we are able to create a symbolic link between the colima socket file and the docker sock path. 

I ran `sudo ln -sf $HOME/.colima/default/docker.sock /var/run/docker.sock` and it worked fine ! 

Thanks for the support anyway. Please let me know if other container runtimes need to be supported. Would be happy to help !

**Comment by lme-nca** [comment]
Yes this works as well, however sadly i'm on a managed machine and this involves me getting the sudo permission first, which is rather cumbersome :)

**Comment by knqyf263** [comment]
I think `DOCKER_HOST` helps, but it sounds reasonable to add a flag.

**Comment by aswath-s-tw** [comment]
In that case, I would like to work on it. Can you assign me to this issue ?

**Comment by knqyf263** [comment]
>In that case, I would like to work on it. Can you assign me to this issue ?

Sure! @AndreyLevchenko will help. I think we want to add `--docker-host`.

**Comment by aswath-s-tw** [comment]
Hi, Can i load `--docker-host` as a value in the struct `DockerOption` located in `pkg/fanal/types/docker.go`

**Comment by aswath-s-tw** [comment]
Hi @knqyf263 , if this approach is fine, I can go ahead with the implementation. Wanted to follow up. What do you think ?

**Comment by aswath-s-tw** [comment]
ping @AndreyLevchenko @knqyf263

**Comment by aswath-s-tw** [comment]
ping @AndreyLevchenko @knqyf263

**Comment by knqyf263** [comment]
Sounds good

**Comment by aswath-s-tw** [comment]
Hi @knqyf263 can I get some ideas or pointers around how I should write tests for this feature ?

**Comment by AndreyLevchenko** [comment]
Hi @aswath-s-tw 
Could you publish PR draft so we could review your code and suggest something related to tests

**Comment by aswath-s-tw** [comment]
Hey @knqyf263 @AndreyLevchenko please review the code and suggest changes if any. Thanks !
